# -*- coding: utf-8 -*-
"""RNASEQ_Take2.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1X1xV1teNSBWriCk3UXY6yQcppYmfNyuk
"""

#This cell connects the python notebook to the Google Drive
from google.colab import drive
drive.mount('/content/drive')

#This cell changes to working directory. Make sure to adjust your working directory.
cd '/content/drive/My Drive/Colab/RNA Seq'

!apt-get install fastqc

#Quality control for fastq.gz files
!fastqc *.fastq.gz

!apt-get install hisat2

#This builds the genomic index from the fna file in the current directory.
# Adjust the index name depending on your organism/genome of interest and make sure there is only one .fna file, which is the fasta file of the reference sequence.
!hisat2-build *.fna CCanadensis_index

#Run the alignment in separate lines for each file, so that if it crashes you don't have to rerun all of them
#This step takes a long time to complete
!hisat2 -x CCanadensis_index -U watera.fastq.gz -S watera_alignment.sam
!hisat2 -x CCanadensis_index -U waterb.fastq.gz -S waterb_alignment.sam

#Same as above.
!hisat2 -x CCanadensis_index -U glyphosatea.fastq.gz -S glyphosatea_alignment.sam
!hisat2 -x CCanadensis_index -U glyphosateb.fastq.gz -S glyphosateb_alignment.sam

!apt-get install samtools

#Converting .sam alignments to .bam files
!samtools view -S -b watera_alignment.sam > watera_alignment.bam
!samtools view -S -b waterb_alignment.sam > waterb_alignment.bam
!samtools view -S -b glyphosatea_alignment.sam > glyphosatea_alignment.bam
!samtools view -S -b glyphosateb_alignment.sam > glyphosateb_alignment.bam

!apt-get install subread

#Creating a counts table, which is the input file for the DESEQ analysis. Make sure there is one gtf file in your working directory, corresponding to the annotated genome of interest.
!featureCounts -a *c.gtf -o counts.txt *.bam